hibernate-reactive-core/src/main/java/org/hibernate/reactive/stage/impl/StageSessionFactoryImpl.java

```diff
@@ -97,6 +105,13 @@ public StageSessionFactoryImpl(SessionFactoryImpl delegate) {
 				.thenApply( s -> new StageStatelessSessionImpl(s, this) ) );
 	}
 
+	CompletionStage<Stage.StatelessSession> newStatelessSession(String tenantId) {
+		return stage( v -> connection( tenantId )
+				.thenApply(
+						connection -> new ReactiveStatelessSessionImpl( delegate, options( tenantId ), connection ) )
```

Comment:
this needs to change to 
```
.thenCompose( connection -> create( reactiveConnection, () -> new ReactiveStatelessSessionImpl( delegate, options( tenantId ), connection ) ) )
```

See the other similar methods for reference.
Otherwise the connection doesn't get closed if an error occurs in the constructor of the session

Reply:
fixed